src/main/java/vinyldns/java/model/batch/AddSingleChange.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright 2018 Comcast Cable Communications Management, LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package vinyldns.java.model.batch;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+import vinyldns.java.model.record.RecordType;
+import vinyldns.java.model.record.data.RecordData;
+
+public class AddSingleChange implements SingleChange {
+  private String id;
+  private String zoneId;
+  private String zoneName;
+  private RecordType type;
+  private String recordName;
+  private String inputName;
+  private SingleChangeStatus status;
+  private String systemMessage;
+  private String recordChangeId;
+  private String recordSetId;
```

Comment:
Since we have the enumeration class for `ChangeInputType`, should we also do something similar for `SingleChangeType`  for `AddSingleChange` and `DeleteRecordSetSingleChange` (see: https://www.vinyldns.io/api/batchchange-model.html#singleaddchange-)?

Reply:
@mitruly 
`ChangeInputType` and `SingleChangeType` are the same enum:
  * Add
  * DeleteRecordSet
https://www.vinyldns.io/api/create-batchchange.html#addchangeinput-attributes
https://www.vinyldns.io/api/batchchange-model.html#singleaddchange-

I don't think that we need to have 2 separated enums.

Reply:
@enikulesku That is a fair point.